tests/manager/models/test_utils.py

```diff
@@ -133,3 +142,240 @@ async def test_agg_to_array(session_info) -> None:
             (kernels.c.tag == test_data2) & (kernels.c.session_id == session_id)
         )
     )
+
+
+@pytest.fixture
+async def dummy_kernels(database_engine: ExtendedAsyncSAEngine):
+    # dummy_kernels, designed solely for testing sql_json_merge, only includes the status_history column, unlike legacy kernels table.
+    dummy_kernels = sa.Table(
+        "dummy_kernels",
+        metadata,
+        sa.Column("id", sa.Integer(), primary_key=True, default=1),
+        sa.Column(
+            "status_history", pgsql.JSONB(), nullable=True, default=sa.null()
+        ),  # JSONB column for testing
+        extend_existing=True,
+    )
+
+    async with database_engine.begin() as db_sess:
+        await db_sess.run_sync(metadata.create_all)
+        await db_sess.execute(dummy_kernels.insert())  # insert fixture data for testing
+        await db_sess.commit()
+
+        yield dummy_kernels
+
+    async with database_engine.begin() as db_sess:
+        await db_sess.run_sync(dummy_kernels.drop)
+        await db_sess.commit()
+
+
+@pytest.mark.asyncio
+async def test_sql_json_merge__deeper_object(
+    dummy_kernels: sa.Table, database_engine: ExtendedAsyncSAEngine
+):
+    async with database_engine.begin() as db_sess:
+        timestamp = datetime.now(tzutc()).isoformat()
+        expected = {
+            "kernel": {
+                "session": {
+                    "PENDING": timestamp,
+                    "PREPARING": timestamp,
+                },
+            },
+        }
+        query = (
+            dummy_kernels.update()
+            .values({
+                "status_history": sql_json_merge(
+                    dummy_kernels.c.status_history,
+                    ("kernel", "session"),
+                    {
+                        "PENDING": timestamp,
+                        "PREPARING": timestamp,
+                    },
+                ),
+            })
+            .where(dummy_kernels.c.id == 1)
+        )
+        await db_sess.execute(query)
+        result = (await db_sess.execute(sa.select(dummy_kernels.c.status_history))).scalar()
+        assert result == expected
```

Comment:
I can't understand what you're trying to test.

Reply:
This PR simply moves the old sql_json_merge test code as it is, making it difficult to understand and including a lot of unnecessary code.

I think It would be better to remove unnecessary `dummy_kernels` and rewrite all these the test codes.